rskj-core/src/main/java/co/rsk/pcc/blockheader/GetBlockHash.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.pcc.blockheader;
+
+import co.rsk.pcc.ExecutionEnvironment;
+import co.rsk.pcc.NativeMethod;
+import org.ethereum.core.Block;
+import org.ethereum.core.CallTransaction;
+import org.ethereum.util.ByteUtil;
+
+import java.math.BigInteger;
+
+/**
+ * This implements the "getBlockHash" method
+ * that belongs to the BlockHeaderContract native contract.
+ *
+ * @author Diego Masini
+ */
+public class GetBlockHash extends NativeMethod {
+    private final BlockAccessor blockAccessor;
+
+    private final CallTransaction.Function function = CallTransaction.Function.fromSignature(
+            "getBlockHash",
+            new String[]{"uint256"},
+            new String[]{"bytes"}
+    );
+
+    public GetBlockHash(ExecutionEnvironment executionEnvironment, BlockAccessor blockAccessor) {
+        super(executionEnvironment);
+        this.blockAccessor = blockAccessor;
+    }
+
+    @Override
+    public CallTransaction.Function getFunction() {
+        return function;
+    }
+
+    @Override
+    public Object execute(Object[] arguments) {
+        short blockDepth = ((BigInteger) arguments[0]).shortValueExact();
```

Comment:
All `BlockHeader` `NativeMethod`s have the same pattern. I wonder if the repeated code could be factored out.

Reply:
One option is having a `BlockHeaderNativeMethod` class, and instead of having one class per method we could parameterize it. E.g:

```java
    @Override
    public List<NativeMethod> getMethods() {
        return Arrays.asList(
                new BlockHeaderNativeMethod(getExecutionEnvironment(), this.blockAccessor, Block:: getHash),
                new BlockHeaderNativeMethod(getExecutionEnvironment(), this.blockAccessor, Block:: getBitcoinMergedMiningHeader),
                ...
        );
    }
```